### ISSUE TITLE
Failed instance is not terminated

### ISSUE THREAD

**Issue by Xerus2000** [sghpjuikit/player#13]
When I launch it and it fails to launch, I get that screen "Application did not start successfully."
But the instance then runs in the background and doesn't terminate itself, which means that the next launched instance fails silently with the following error:

```
50:09.453 I [JavaFX-Launcher] App - App will close prematurely: Multiple app instances detected
50:09.620 I [JavaFX Applicat] App - Application closing prematurely
Exception in Application stop method
50:09.906 E [           main] CoreLogging - Uncaught exception
java.lang.reflect.InvocationTargetException: null
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:564) ~[na:na]
	at java.base/sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:945) ~[na:na]
Caused by: java.lang.RuntimeException: Exception in Application stop method
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:978) ~[javafx.graphics:na]
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:198) ~[javafx.graphics:na]
	at java.base/java.lang.Thread.run(Thread.java:844) ~[na:na]
Caused by: kotlin.UninitializedPropertyAccessException: lateinit property executor has not been initialized
	at sp.it.pl.core.CoreSerializer.dispose(CoreSerializer.kt:28) ~[PlayerFX/:na]
	at sp.it.pl.main.App.stop(App.kt:410) ~[PlayerFX/:na]
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$10(LauncherImpl.java:938) ~[javafx.graphics:na]
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$11(PlatformImpl.java:449) ~[javafx.graphics:na]
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$9(PlatformImpl.java:418) ~[javafx.graphics:na]
	at java.base/java.security.AccessController.doPrivileged(Native Method) ~[na:na]
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:417) ~[javafx.graphics:na]
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96) ~[javafx.graphics:na]
	at javafx.graphics/com.sun.glass.ui.win.WinApplication._runLoop(Native Method) ~[javafx.graphics:na]
	at javafx.graphics/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:189) ~[javafx.graphics:na]
	... 1 common frames omitted
```

I think there should be a simple message like "There is already an instance running!" and of course the Application should terminate itself when it fails.

**Comment by sghpjuikit** [comment]
1 The application has a mechanism to prevent multiple instances, but it seems to fail in some cases, I'll look into it.

2 I agree that if the application fails to launch, it should close gracefully after user hits OK or something.

**Comment by sghpjuikit** [comment]
1) The issue is that I ported _App.class_ to kotlin and JavaFX had trouble instantiating, so I made _AppUtil.class_ just to hold _public static void main_ method, which changes the name of the entry class, through which the application is started. The detection of number of application instances running is done by inspecting all running JVMs and identifying them via the name of this entry class. I forgot to change the name in the comparison after I changed the entry class, which caused the getInstanceCount() always return zero.

2) I'll improve the Application start failure message by including information about the existence of log, containing the problem, or perhaps include the exception in the message itself. I'll also make it so the application shuts down gracefully after user closes the error message.

3) The stacktrace you posted is caused by exception raised during disposing a component - this happens because the component was not initialized (due to some problem while the application starts). This can be fixed by mandating a contract for any _Core_ component to make no assumptions in its _dispose()_ method about whether its _init()_ method was invoked. It's a design issue I overlooked, but nothing problematic, it simply complicates the dispose method and must be explicitly stated in javadoc to avoid this issue in the future.

Thanks for reporting this, I'll fix this soon enough.
